src/unread/__tests__/unreadModel-test.js

```diff
@@ -63,6 +67,114 @@ describe('stream substate', () => {
     });
   });
 
+  describe('EVENT_UPDATE_MESSAGE', () => {
+    const mkAction = args => {
+      const {
+        message_ids,
+        stream_id = 123,
```

Comment:
I think this default parameter means that `mkAction`'s callers can't request an action that omits `stream_id`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#passing_undefined_vs._other_falsy_values

Whether those callers will end up wanting to do that will depend on whether servers want to convey something relevant by omitting `stream_id`, I guess. You've written down the state of that question in `unreadModel`:

```js
      if (stream_id == null) {
        // Not stream messages, or else a pure content edit (no stream/topic change.)
        //
        // As of this writing, the docs actually promise this field for all
        // updates to stream messages.  Empirically it's absent on pure
        // content edits, but present on edits affecting either the stream
        // or topic:
        //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60update_message.60.20event/near/1296823
        return state;
      }
```

Reply:
True. At present this test file is just for the "streams" part of the unreads data, because that's the part that has been modernized. In the future when we similarly modernize the rest and have its tests in the same place, we'll want to either add a different helper for those tests or modify this one.

Reply:
Makes sense. Also, I see from your other comment that we've settled that servers will start always including `stream_id` for stream messages, so there's no reason for this modernized streams-focused part of the model to be tested with an event that's missing `stream_id`.